agent/structs/acl.go

```diff
@@ -10,7 +11,7 @@ import (
 
 	"github.com/hashicorp/consul/acl"
 	"github.com/hashicorp/consul/sentinel"
-	"github.com/mitchellh/hashstructure"
+	"golang.org/x/crypto/blake2b"
```

Comment:
Reasons for using blake2b were:

1. I ripped a fair amount of the hashing code direct from nomad and they use blake2b.
2. Various benchmarks show its faster to compute than SHA256

Maybe we could ask Armon if there are other advantages as he was the one who did it for Nomad:
https://github.com/hashicorp/nomad/commit/84d87948f6402cf2a7645808fc3c9f79628b7562

Really after going back and forth on this a few times I am fairly certain a cryptographic hash isn't entirely necessary (at least not with the present state). Regardless I ripped off the blake2b hashing of policies and tokens from

Reply:
I'm happy with blake - it has a stronger crypto properties than sha256 and if we already do it in nomad then cool :) We should maybe converge on this in other places too. Not necessary to go replace sha256 everywhere but just to avoid having tons of different standards etc.